### DISCUSSION TITLE
About NaNs appearing in unet shortly after a couple of generations. This lingers for quite some time already.

### DISCUSSION THREAD

**Discussion by mart-hill** [Bing-su/adetailer#114] [Q&A]
Hi!
I'd like to ask, which model should I use to avoid that? Because normally, after just enabling the extension with the default face_yolov8n.pt model (others usually make it only worse) I'm getting NaNs fairly quickly (3-4 "Generate" button presses). I'm using FP16 with --no-half-vae on Torch 2.0.1 and RTX 3090 (Vlad's fork on Windows 10 22H2). VAE is vae-ft-mse-840000(...).
I'm not even calling ControlNet within the extension (this would only worsen the issue).

Am I doing something wrong?

Edit: I'm observing a change (sudden NaN "disappearance") after changing just one letter in `<lyco:shouyouHinataHaikyuu_v2:0.8>` (first 's' should be CAPS, this is the LoRA's filename). In other words, "disabling" one LoRA can change the NaN behavior. Why?

**Comment by Bing-su** [comment]
The adetailer is not connected directly to the image generation. It is entirely up to the checkpoint, VAE, lora, etc. used to generate the image.

In your case, it means that the lora you used was fine with txt2img, but caused problems with inpainting, so you should contact the creator of that lora, not here.

**Reply by mart-hill** [reply]
Ohh, so that's what's going on. I thought the option to "upcast" the calculations in the adetailer would help. :)
The funny thing is that re-enabling this LoRA created the image without issue. Could the LoRA be selectively/intermittently behaving bad with inpainting one time, and then not?

Thanks for the help!

**Comment by mart-hill** [comment]
I'm testing the following settings for Compute Settings and Token Merging in Vlad's fork, those aren't giving me NaNs (so that one Compute option concerning NaNs being ON is just a safeguard). All on Windows 10 22H2, RTX 3090, Torch 2.1 nightly and **Scaled-Dot-Product** as cross-attention optimization method.

Compute Settings:
![image](https://github.com/Bing-su/adetailer/assets/55754730/8e7c4635-91e4-49a5-8286-2d5dcd0b7376)

Token Merging:
![image](https://github.com/Bing-su/adetailer/assets/55754730/9d791558-b03b-48bf-bea8-50d43562bfe0)
